worldedit-core/src/main/java/com/boydti/fawe/FaweAPI.java

```diff
@@ -344,24 +348,29 @@ public static int fixLighting(World world, Region selection, @Nullable IQueueExt
             }
         }
 
-        NMSRelighter relighter = new NMSRelighter(queue, Settings.IMP.LIGHTING.DO_HEIGHTMAPS);
-        for (int x = minX; x <= maxX; x++) {
-            for (int z = minZ; z <= maxZ; z++) {
-                relighter.addChunk(x, z, null, 65535);
-                count++;
+        try (Relighter relighter = WorldEdit.getInstance().getPlatformManager().queryCapability(Capability.WORLD_EDITING)
+                .getRelighterFactory()
+                .createRelighter(mode, world, queue)) {
+
+            for (int x = minX; x <= maxX; x++) {
+                for (int z = minZ; z <= maxZ; z++) {
+                    relighter.addChunk(x, z, null, 65535);
+                    count++;
+                }
             }
-        }
-        if (mode != RelightMode.NONE) {
-            if (Settings.IMP.LIGHTING.REMOVE_FIRST) {
-                relighter.removeAndRelight(true);
+            if (mode != RelightMode.NONE) {
+                if (Settings.IMP.LIGHTING.REMOVE_FIRST) {
+                    relighter.removeAndRelight(true);
+                } else {
+                    relighter.fixSkyLighting();
+                    relighter.fixBlockLighting();
+                }
             } else {
-                relighter.fixSkyLighting();
-                relighter.fixBlockLighting();
+                relighter.removeLighting();
             }
-        } else {
-            relighter.removeLighting();
+        } catch (Exception e) {
+            LOGGER.error("Error occurred on fix lighting", e);
```

Comment:
Might be worth being specific here to say we couldn't find a relighter?

Reply:
The exception is part of the signature of AutoClosable#close() - I just thought that would look a bit better, but it doesn't really matter to me

Reply:
I meant saying something like `Error occurred attempting to initialise relighter. Report to ....`

Reply:
The exception can occur at any point during relighting, so I think that would be rather confusing